src/contracts/voting/VotingMachine.sol

```diff
@@ -184,4 +116,60 @@ contract VotingMachine is VotingMachineWithProofs, IVotingMachine {
 
     emit GasLimitUpdated(gasLimit);
   }
+
+  function _checkOrigin(
+    address caller,
+    address originSender,
+    uint256 originChainId
+  ) internal view override {
+    require(
+      caller == CROSS_CHAIN_CONTROLLER &&
+        originSender == L1_VOTING_PORTAL &&
+        originChainId == L1_VOTING_PORTAL_CHAIN_ID,
+      Errors.WRONG_MESSAGE_ORIGIN
+    );
+  }
+
+  /// @dev creates a proposal vote
+  function _parseReceivedMessage(
+    address originSender,
+    uint256 originChainId,
+    BridgingHelper.MessageType messageType,
+    bytes memory message
+  ) internal override {
+    bytes memory empty;
+    if (messageType == BridgingHelper.MessageType.Proposal_Vote) {
+      try this.decodeStartProposalVoteMessage(message) returns (
+        uint256 proposalId,
+        bytes32 blockHash,
+        uint24 votingDuration
+      ) {
+        _createBridgedProposalVote(proposalId, blockHash, votingDuration);
+        emit MessageReceived(
```

Comment:
maybe we can abstract this part as well, checking that msg.sender is address(this)

Reply:
Did not find a way on abstracting, as condition is specific on the inheriting contract